runtime/js/12_io.js

```diff
@@ -98,29 +95,16 @@
     }
 
     const nread = await core.binOpAsync("op_read_async", rid, buffer);
-    if (nread < 0) {
-      throw new Error("read error");
-    }
 
     return nread === 0 ? null : nread;
   }
 
   function writeSync(rid, data) {
-    const result = core.binOpSync("op_write_sync", rid, data);
-    if (result < 0) {
-      throw new Error("write error");
-    }
```

Comment:
I don't understand reason for this change, does "bin op" no longer return negative number in case of error?

Reply:
In case of an error, an error is thrown, so nothing is returned.

Reply:
I think this was also how "minimal ops" worked.

Reply:
Makes sense, thanks for explanation